### PR TITLE
Load modules

### PR DESCRIPTION
**Pull request by thekid**
This pull request implements xp-framework/rfc#306 and adds `-m [module]`. The `module` argument references a module by its qualified name `[vendor]/[package]`. If module is an existent directory, all path files inside it are scanned for class path entries. Otherwise, the `module` path defined via `xp.ini` is scanned. It defaults to [$COMPOSER_HOME](https://getcomposer.org/doc/03-cli.md#composer-home)/vendor/{vendor}/{name}.

### CODE REVIEW

**Comment by thekid** [comment]
I'm not all too happy with this implementation - it adds quite a bit of complexity. I would rather split this in two parts:

## Module paths
First, make the following work:

```sh
$ xp -m ../unittest/ -m ../logging/ xp.unittest.Runner src/test/config/unittest/core.ini
```

...as a short form of:

```sh
$ xp -cp ../unittest/src/main/php -cp ../io-collections/src/main/php -cp ../logging/src/main/php xp.unittest.Runner src/test/config/unittest/core.ini
```

*The io-collections module is a transitive dependency of unittest*

## Module search
The next thing can be to create a module registry / lookup and the ability to supply `vendor/name`-style names on the command line, integrated with Composer or not. 

* * * 

Basically, we're tendering to two usecases here:

1. **Run XP tools from any directory** - making this easier is solved by the first step
2. **Running XP projects** - Here, our progress is actually starting to conflict with Composer. If we have a local registry, why should we run `composer install`? More thought needs to be put into this!
